### PR TITLE
added yml/json/toml support, --help flag, and example directory

### PR DESCRIPTION
**Pull request by tscholl2**

### CODE REVIEW

File: cmd/task/task.go [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+      "echo \"I am going to write a file named 'output.txt' now.\"",
+      "echo \"hello\" > output.txt"
+    ],
+    "generates": [
+      "output.txt"
+    ]
+  }
+}
+'''
+`)
```
Comment (andreynering) [inline cmd/task/task.go]: I think this example should be in YAML.

YAML will be considered the recommended default, in examples and documentation.

File: example/Taskfile.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "hello": {
+    "cmds": [
+      "echo \"I am going to write a file named 'output.txt' now.\"",
+      "echo \"hello\" > output.txt"
+    ],
+    "generates": [
+      "output.txt"
+    ]
+  }
+}
```
Comment (andreynering) [inline example/Taskfile.json]: Missing line break.

Comment (andreynering) [inline example/Taskfile.json]: Also missing TOML example.

Comment (tscholl2) [inline example/Taskfile.json]: Done. I don't know if these examples really highlight the parts of task that you want them to, so you might consider changing them. But I hope this helps some.

**Comment by andreynering** [comment]
Thank you again.

Merged as https://github.com/go-task/task/commit/61b1aa8559dda4469dae5579b3bbc17f1c4683c7 and https://github.com/go-task/task/commit/a53fcf8bbac78616906db9444826587297b34220

**Pull request closed without merging**